### PR TITLE
Replace plaintext License with Markdown variant

### PR DESCRIPTION
**Pull request by sdulal**
The change in this pull request includes a license that is better formatted for viewing on GitHub and some other systems.

### CODE REVIEW

**Comment by JonRowe** [comment]
Hi, I think I'm generally in favour of this change, but if we're going to do this we should do it on rspec/rspec-dev so that the file is the same across all the RSpec gems and repos, would you like to open a PR there for this?

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
+
```
Comment (JonRowe) [inline LICENSE.md]: Also we generally don't have blank lines at the end of our files.

File: LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+The MIT License (MIT)
+=====================
+
+Copyright © `2009` `Chad Humphries, David Chelimsky`  
+Copyright © `2006` `David Chelimsky, The RSpec Development Team`  
+Copyright © `2005` `Steven Baker`
```
Comment (JonRowe) [inline LICENSE.md]: There's no need to code quote these details, text is fine.

**Comment by myronmarston** [comment]
> Hi, I think I'm generally in favour of this change, but if we're going to do this we should do it on rspec/rspec-dev so that the file is the same across all the RSpec gems and repos, would you like to open a PR there for this?

The file should not be the same across all RSpec gems and repos.  The different gems and repos have different copyright notices due to their different history.  Compare:
- https://github.com/rspec/rspec-core/blob/master/License.txt
- https://github.com/rspec/rspec-rails/blob/master/License.txt (this one should probably be updated to list @cupakromer since he's the maintainer now)
- https://github.com/rspec/rspec-support/blob/master/LICENSE.txt

The text of the MIT license doesn't need to change much, either (in all my years of open source work I've never updated the MIT license wording in any project) so I don't think we need an automated way to keep that in sync.

**Pull request merged**